internal/controller/cloud-resources/aws/awsnfsbackupschedule_test.go

```diff
@@ -4,67 +4,94 @@ import (
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
-	cloudresourcesv1beta1 "github.com/kyma-project/cloud-manager/api/cloud-resources/v1beta1"
 	. "github.com/kyma-project/cloud-manager/pkg/testinfra/dsl"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// I can not make this work, good luck figuring this one
-var _ = SkipDescribe("Feature: SKR AwsNfsBackupSchedule", func() {
+var _ = Describe("Feature: SKR AwsNfsBackupSchedule", func() {
 
 	const (
-		interval = time.Millisecond * 50
-	)
-	var (
-		timeout = time.Second * 20
+		interval        = time.Millisecond * 50
+		timeout         = time.Second * 20
+		awsAccountId    = "974658265573"
+		toleranceWindow = 120 * time.Second
 	)
-	var now time.Time
-	var skrNfsVolumeName string
-	var skrNfsVolume *cloudresourcesv1beta1.AwsNfsVolume
-	var nfsInstanceName string
-	var nfsInstance *cloudcontrolv1beta1.NfsInstance
-	var awsNfsId string
-	var scope *cloudcontrolv1beta1.Scope
 
-	awsAccountId := "974658265573"
+	It("Scenario: Creates recurring backup schedule with existing backups", func() {
+		var (
+			suffix             string
+			skrNfsVolumeName   string
+			nfsInstanceName    string
+			scheduleName       string
+			existingBackupName string
+			skrNfsVolume       *cloudresourcesv1beta1.AwsNfsVolume
+			nfsInstance        *cloudcontrolv1beta1.NfsInstance
+			scope              *cloudcontrolv1beta1.Scope
+			backupSchedule     *cloudresourcesv1beta1.AwsNfsBackupSchedule
+			existingBackup     *cloudresourcesv1beta1.AwsNfsVolumeBackup
+		)
 
-	BeforeEach(func() {
-		By("Given KCP Scope exists", func() {
-			now = time.Now().UTC()
-			skrNfsVolumeName = uuid.NewString()
-			skrNfsVolume = &cloudresourcesv1beta1.AwsNfsVolume{}
-			nfsInstanceName = uuid.NewString()
-			nfsInstance = &cloudcontrolv1beta1.NfsInstance{}
-			awsNfsId = uuid.NewString()
-			scope = &cloudcontrolv1beta1.Scope{}
+		// This ensures each parallel test run has different resource names
+		suffix = fmt.Sprintf("recurring-%d", GinkgoParallelProcess())
+		skrNfsVolumeName = fmt.Sprintf("aws-vol-%s", suffix)
+		nfsInstanceName = fmt.Sprintf("aws-inst-%s", suffix)
+		scheduleName = fmt.Sprintf("aws-schedule-%s", suffix)
+		existingBackupName = fmt.Sprintf("aws-backup-existing-%s", suffix)
+
+		// Initialize objects
+		skrNfsVolume = &cloudresourcesv1beta1.AwsNfsVolume{}
+		nfsInstance = &cloudcontrolv1beta1.NfsInstance{}
+		scope = &cloudcontrolv1beta1.Scope{}
+		backupSchedule = &cloudresourcesv1beta1.AwsNfsBackupSchedule{}
+		existingBackup = &cloudresourcesv1beta1.AwsNfsVolumeBackup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      existingBackupName,
+				Namespace: DefaultSkrNamespace,
+			},
+		}
+
+		DeferCleanup(func() {
+			By("Cleanup: Delete test resources", func() {
+				_ = Delete(infra.Ctx(), infra.SKR().Client(), backupSchedule)
```

Comment:
We do no want to ignore errors
```suggestion
				Expect(Delete(infra.Ctx(), infra.SKR().Client(), backupSchedule)).To(Succeed())
```
Ideally you would also ensure it was really deleted and not leaking out of scenario with
```suggestion
				Eventually(IsDeleted).WithArguments(infra.Ctx(), infra.SKR().Client(), backupSchedule).Should(Succeed())
```